scripts/screeps_rl_live_dashboard.py

```
@@ -0,0 +1,806 @@
+            payload = load_json_object(path)
+            if payload is None:
+                continue
+            candidates.append((path, payload, artifact_timestamp(path, payload)))
+    if not candidates:
+        return None
+    return max(candidates, key=lambda item: item[2])
+
+
+def db_latest_timestamp(conn: sqlite3.Connection, table_name: str, column_name: str) -> str | None:
+    try:
+        row = conn.execute(f"SELECT MAX({column_name}) FROM {table_name}").fetchone()
```

Ruff (0.15.13)

[error] Possible SQL injection vector through string-based query construction

(S608)